tests/functional/test_new_resolver_target.py

```diff
@@ -0,0 +1,74 @@
+import pytest
+
+from pip._internal.cli.status_codes import ERROR, SUCCESS
+from tests.lib.path import Path
+from tests.lib.wheel import make_wheel
+
+
+@pytest.fixture()
+def fake_wheel_tag():
+    return "fakepy1-fakeabi-fakeplat"
+
+
+@pytest.fixture()
+def fake_wheel(script, fake_wheel_tag):
```

Comment:
nit: it may be a bit cleaner to have our `fake_wheel` fixture return a function that takes the fake_wheel_tag, so that we can put the fake wheel tag in our test function, closer to the related test data.

Reply:
I’m not following the last sentence 🤯

Reply:
Like

```
@pytest.fixture()
def make_fake_wheel(script);
    def _make_fake_wheel(fake_wheel_tag):
        # current contents of fake_wheel
        ...
    return _make_fake_wheel

def test_new_resolver_target_checks_compatibility_failure(
    script,
    make_fake_wheel,
    ...
):
    fake_wheel = make_fake_wheel("fakepy1-fakeabi-fakeplat")
    ...
```

Then the fake wheel tag is only a few lines away from the related params in `pytest.parametrize` instead of 15.